docs/decisions/0015-expiration-improvements.rst

```diff
@@ -0,0 +1,94 @@
+0015 Expiration and Cancellation Data Improvements
+**************************************************
+
+Status
+======
+Proposed - December 2023
+
+Context
+=======
+There are subtleties around the Assignment cancellation and expiration user
+experience in the enterprise learner-portal frontend that require a more nuanced
+and descriptive approach to the data we persist around these states, and
+how that data is represented in API response payloads for consumption
+by the learner-portal frontend.
+
+How expiration currently works
+------------------------------
+There's currently a management command (run via a cron) that toggles the state
+of Assignments to ``CANCELLED`` under the following conditions:
+
+1. The current date is more than 90 days after the *creation* date of the assignment. We use the creation
+   date as a proxy for the date at which an assignment was allocated, which is problematic under
+   the edge case where an assignment is allocated, cancelled, and re-allocated later.
+2. The current date is greater than the inferred enrollment deadline for the assigned course.
+3. The current date is greater than the expiration date of the subsidy associated with the Assignment's
+   access policy record.
+
+Learners are notified of expiration and cancellation via the learner-portal frontend
+------------------------------------------------------------------------------------
+The learner-portal notifies learners of assignment cancellation and expiration.  These notifications
+are required to be dismissable, which means we need to persist state about dismissal
```

Comment:
nit: Is it worth defining "dismissable" a bit more explicitly? That is, _permanently_ dismissable where after seeing/dismissing them in the UI, the learner would never seen them again (removes any ambiguity around whether "dismiss" means for the page view, for the session, forever, etc.).